api/client/client.go

```diff
@@ -3,3 +3,74 @@
 // Run "docker help SUBCOMMAND" or "docker SUBCOMMAND --help" to see more information on any Docker subcommand, including the full list of options supported for the subcommand.
 // See https://docs.docker.com/installation/ for instructions on installing Docker.
 package client
+
+import (
+	"io"
+
+	"github.com/docker/docker/api/client/lib"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/cliconfig"
+	"github.com/docker/docker/pkg/parsers/filters"
+	"github.com/docker/docker/registry"
+	"github.com/docker/docker/runconfig"
+)
+
+// apiClient is an interface that clients that talk with a docker server must implement.
+type apiClient interface {
+	ContainerAttach(options types.ContainerAttachOptions) (*types.HijackedResponse, error)
```

Comment:
Does this have to return pointer? + few other places (ExecStartCheck, ExecConfig). I think if there is no strict requirement for it we should be consistent with other methods.